### PR TITLE
Implement reminders

### PR DESCRIPTION
**Pull request by vicky5124**
Aquest PR implementa recordatoris singulars de temps relatiu.
També implementa migracions de bases de dades, ja que són necessàries per a implementar el principal.

TODO:
- [x] Switch from sqlite to cassandra.
- [x] Fix the random performance.
- [x] Fix all the pyright `Cannot access member "..." for type "Session"`
- [x] Add documentation on how to run the bot with the new 3rd party software required.

### CODE REVIEW

File: src/plugins/reminders.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        time += int(match.group("days")) * 86400
+    if match.group("hours"):
+        time += int(match.group("hours")) * 3600
+    if match.group("minutes"):
+        time += int(match.group("minutes")) * 60
+    if match.group("seconds"):
+        time += int(match.group("seconds"))
+
+    if time < 60:
+        await ctx.respond("El temps del recordatori es massa curt.")
```
Comment (helena-dev) [inline src/plugins/reminders.py:74]: typo -> és

File: src/plugins/reminders.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            message_id,
+            ctx.channel_id,
+            ctx.guild_id,
+            time,
+            ctx.options.message or None,
+        ),
+    )
+
+    await ctx.respond(
+        f"Recordatori guardat! El recordatori sera enviat <t:{int(time.timestamp())}:F>!\nID: {reminder_id}"
```
Comment (helena-dev) [inline src/plugins/reminders.py:101]: typo, falta accent. "serà"

File: src/plugins/reminders.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+async def delete_reminder(ctx: utils.Context) -> None:
+    """
+    Deletes a reminder.
+    """
+    if not ctx.options.reminder_id:
+        await ctx.respond("No he pogut entendre l'ID del recordatori.")
+        return
+
+    if len(ctx.options.reminder_id) != 8:
+        await ctx.respond("La ID es invalida.")
```
Comment (helena-dev) [inline src/plugins/reminders.py:120]: typo, "és"

File: src/plugins/reminders.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+async def reminder_info(ctx: utils.Context) -> None:
+    """
+    Shows information about a reminder.
+    """
+    if not ctx.options.reminder_id:
+        await ctx.respond("No he pogut entendre l'ID del recordatori.")
+        return
+
+    if len(ctx.options.reminder_id) != 8:
+        await ctx.respond("La ID es invalida.")
```
Comment (helena-dev) [inline src/plugins/reminders.py:189]: typo, "és"

File: src/plugins/reminders.py [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+async def delete_reminder(ctx: utils.Context) -> None:
+    """
+    Deletes a reminder.
+    """
+    if not ctx.options.reminder_id:
+        await ctx.respond("No he pogut entendre la ID del recordatori.")
+        return
+
+    if len(ctx.options.reminder_id) != 8:
+        await ctx.respond("La ID és invà-lida.")
```
Comment (vicky5124) [inline src/plugins/reminders.py:120]:
```suggestion
        await ctx.respond("La ID és invàlida.")
```

**Pull request merged**